### ISSUE TITLE
Add support for Control components

### ISSUE THREAD

**Issue by nitaliano** [nitaliano/react-native-mapbox-gl#977]
This will enable users to place custom views on top of the map with some built in position styles to make it nice and easy. This will also allows us to have a common ui library for common buttons, views, and map layers

**Comment by HikingDev** [comment]
Wow, this is awesome news!
If I can be of any help let me know

**Comment by EricPKerr** [comment]
@nitaliano will this support the ability to customize views like the native UserLocation view?  Know it's possible to do this with our own components (like [this](https://github.com/mapbox/react-native-mapbox-gl/blob/master/example/src/components/common/PulseCircleLayer.js)), but would be nice to get some of the `userTrackingMode` and `userLocationVerticalAlignment` functionality with a component that matches our own styling and UI.  Thanks!

**Comment by nitaliano** [comment]
Yes, that is eventually what is going to happen, we have plans on adding a `UserLocation` component that you will be able to customize completely using layers

**Comment by onitzschke** [comment]
+1

**Comment by onitzschke** [comment]
+1

**Comment by ippa** [comment]
@nitaliano  Is this dev-push still active? :) would be really nice to have on android.

**Comment by sheeshmohsin** [comment]
+1

**Comment by DavidKHahn** [comment]
is this still active?

**Comment by EricPKerr** [comment]
This is all in the forked https://github.com/react-native-mapbox-gl/maps repo

**Comment by DavidKHahn** [comment]
@EricPKerr I'm still getting an 'ReactTextView cannot be cast to android.view.ViewGroup'. I checked out #953 already and tried replacing the View tags outside from Mapview.  Is this also forked into the new repo (currently using it)?

**Comment by EricPKerr** [comment]
You should post issues on the new repository.  As far as I know, the one @nitaliano maintained here is completely deprecated.

**Comment by DavidKHahn** [comment]
@EricPKerr thanks for the quick response times: I am actually using the newer version. I am still running into the same issue however and a google search led me to find a similar issue referenced on this page (#953 which is associated with #977).
